wireapp/wire-android-reloaded
# fix: download and delete from gallery [AR-1149]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1149" title="AR-1149" target="_blank"><img alt="Subtask" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10816?size=medium" />AR-1149</a>  Create Detailed Image Preview screen 
  </td></table>
  <br />
 
















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

After implementing the `MediaGalleryScreen`, QA identified some unimplemented features. These ones are the `Save` and the `Delete` options from the more options dropdown menu.

### Solutions

For the Download method I reused the `FileManager.saveToExternalStorage` method and created a new sealed class called `MediaGallerySnackbarMessages` to feed the snackbar with mapped message codes.

Regarding the deletion mechanism, this was a bit trickier, as I had to create and pass an extra wrapper object, `MessageDeletion` on the `Conversation` navigation item class in order to be able to reuse the deletion mechanism of the `ConversationScreen`. 

### Testing

#### Test Coverage (Optional)


#### How to Test

Open any conversation, send or receive an image message. Click on the image. Check the image can be downloaded and deleted only for me or for everyone through the `MediaGalleryScreen`.

### Attachments (Optional)

https://user-images.githubusercontent.com/2468164/170480400-ef9f8d20-01f3-406c-9a97-54a45c057a8d.mov

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
